frontend/coffee/global.coffee

```diff
@@ -116,7 +120,7 @@ createCookie = (name, value, days) ->
         expires = "; expires=" + date.toGMTString()
     else
         expires = "; expires=session"
-    document.cookie = name + "=" + value + expires + "; path=/"
+    document.cookie = name + "=" + value + expires + "; path=/; SameSite=Strict"
```

Comment:
I had to look this up, but are you sure `Strict` is what we want here? It seems to relate to "deep linking" rather than whether cookies could be sent to other sites:

https://web.dev/samesite-cookies-explained/

> When the user is on your site, then the cookie will be sent with the request as expected. However when following a link into your site, say from another site or via an email from a friend, on that initial request the cookie will not be sent. This is good when you have cookies relating to functionality that will always be behind an initial navigation, such as changing a password or making a purchase, but is too restrictive for promo_shown. If your reader follows the link into the site, they want the cookie sent so their preference can be applied.
> 
> That's where SameSite=Lax comes in by allowing the cookie to be sent with these top-level navigations.

The three cookies set via this function are `first_visit`, `dismissed_donation`, and `do-not-offer-registration`. If I click a link from somewhere else (say, CKAN or the status page), I would think we would still want those cookies to be sent.

Reply:
Whoops, that wasn't meant to go into this branch. But thanks for catching it and the info.
I experimented with those a bit, primarily because Firefox shows these warnings:
> Cookie “session” will be soon rejected because it has the “SameSite” attribute set to “None” or an invalid value, without the “secure” attribute. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite

...and the session cookie getting rejected sounds really bad, I suspect it would make users not staying logged in anymore and unable to upload mods and so on.

Sounds like `Lax` would be the correct choice. Regardless, we should probably also set `Secure` to make sure our cookies only get sent over HTTPS (SpaceDock does redirect HTTP to HTTPS anyway). If I read the MDN docs correctly, local testing should be unaffected:
> Cookie is only sent to the server when a request is made with the https: scheme (except on localhost), and therefore is more resistent to man-in-the-middle attacks. 
For Firefox, the https: requirements are ignored when the Secure attribute is set by localhost (since Firefox 75).

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Secure

Unfortunately I'm not able to find any documentation on how Chrome handles those on localhost, but I hope they do the same.

In any way this is probably better placed in a separate PR where we could research and discuss this on its own, so I'm going to remove that change from here.

Reply:
Ahh, I see. For what it's worth, it looks like the session cookie would be set by Flask's authentication framework on the server rather than this client script function. The client usually is not trusted to set whether it is logged in. :wink:

Reply:
Apparently there are some config parameters that Flask uses:

https://blog.miguelgrinberg.com/post/cookie-security-for-flask-applications

Reply:
Removed from this PR, I'll try out that app config setting, and maybe read some more about this cross-origin cookie stuff and what browsers plan to do with them, and put it in a new PR.